src/custom_hooks/useUserData.jsx

```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+import { useReducer, useState } from "react";
+
+export const useUserData = () => {
+  const encodedToken = localStorage.getItem("token");
+  const initialisUserLoggedIn = encodedToken !== null ? true : false;
```

Comment:
changes: it should be " initialIsUserLoggedIn " instead of " initialisUserLoggedIn "